drivers/builtin/src/entropy.c

```diff
@@ -654,7 +654,7 @@ int mbedtls_entropy_self_test(int verbose)
         }
     }
 
-#if defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+#if defined(MBEDTLS_PLATFORM_GET_ENTROPY_ALT)
     if ((ret = mbedtls_entropy_source_self_test(0)) != 0) {
         goto cleanup;
     }
```

Comment:
Missing: tests. I'd expect `test_suite_entropy` to have at least a test for when `mbedtls_platform_get_entropy_alt` returns an error code, and one for when it returns less entropy than expected.